crates/controller/src/selection/bookkeeping.rs

```diff
@@ -187,15 +185,13 @@ fn selected_system(
 
 fn deselected_system(
     mut events: EventReader<DeselectedEvent>,
-    mut markers: Query<&mut CircleMarker>,
+    mut markers: Query<&mut MarkerVisibility>,
     mut bars: EventWriter<UpdateBarVisibilityEvent>,
     mut poles: EventWriter<UpdatePoleVisibilityEvent>,
 ) {
     for event in events.iter() {
-        if let Ok(mut marker) = markers.get_mut(event.0) {
-            marker
-                .visibility_mut()
-                .update_visible(SELECTION_BAR_ID, false);
+        if let Ok(mut visibility) = markers.get_mut(event.0) {
```

Comment:
Ditto what do we do on error?
Could do something like log it using `warn!()` or `error!()` at least.

Reply:
Please see my other comment. We don't want to make strong assumptions about timing of when the event was sent (or the event leading to this event) and about other systems not despawning the entity in the meantime. I would not even log a warning. Maybe a trace!?

Reply:
Ok, I was just wondering.